include/pybind11/detail/common.h

```diff
@@ -154,6 +154,11 @@
 // C4505: 'PySlice_GetIndicesEx': unreferenced local function has been removed (PyPy only)
 #  pragma warning(disable: 4505)
 #  if defined(_DEBUG) && !defined(Py_DEBUG)
+// fix for vs2022: https://github.com/pybind/pybind11/issues/3477
+#    include <crtdefs.h>
+// python 2.7 header uses 'mbstowcs' function
```

Comment:
Oh, if this is actually only needed for Python 2.7 (?), an `#ifdef` here would be best. That way we will know for sure that we can remove this code again when we purge Python 2 support in a couple months.
With the `#ifdef`, it definitely looks good to me. (Even without, but only if Python 3 actually needs it.)

Reply:
No, see the comment I forget to send above. :)